### ISSUE TITLE
[Bug]: Not providing a model for the AzureOpenAI llm causes unexpected context size

### ISSUE THREAD

**Issue by mve** [run-llama/llama_index#13259]
### Bug Description

When no model name is specified when creating an instance of AzureOpenAI a default is chosen which may differ from the model configured in the deployment in Azure. This can cause unexpected behaviour as the context size is chosen based on the model name.

I think model should not have a default value and if non is entered it should show a clear error warning that the model must be provided and that is must be the same name as the one selected in the Azure deployment.

Also inside the class there is the following comment:

```
- model: Name of the model (e.g. `text-davinci-003`)
        This in only used to decide completion vs. chat endpoint.
```
        
I believe this is not accurate as it is also used for the context size.

### Version

0.10.31

### Steps to Reproduce

Create an instance of AzureOpenAI like so:

```
    return AzureOpenAI(
        engine=engine,
        temperature=temperature,
        timeout=timeout,
        # model was not added here.
        api_key=AZURE_OPENAI_API_KEY,
        azure_endpoint=AZURE_OPENAI_API_ENDPOINT,
        api_version=AZURE_OPENAI_API_VERSION,
        max_tokens=max_tokens,
    )
```    

If no model is specified the default of "gpt-35-turbo" is used. If your deployment uses a different model, this can cause unexpected behaviour as the context size will be set to 4096 even though you may have selected a model in Azure that has a larger context size.
    
I think this context size gets chosen based on the model name in this code:

```
    @property
    def metadata(self) -> LLMMetadata:
        return LLMMetadata(
            context_window=openai_modelname_to_contextsize(self._get_model_name()),
            num_output=self.max_tokens or -1,
            is_chat_model=is_chat_model(model=self._get_model_name()),
            is_function_calling_model=is_function_calling_model(
                model=self._get_model_name()
            ),
            model_name=self.model,
        )
```

### Relevant Logs/Tracbacks

```shell
File “/usr/local/lib/python3.12/site-packages/llama_index/core/memory/chat_memory_buffer.py”, line 109, in get
raise ValueError(“Initial token count exceeds token limit”)
ValueError: Initial token count exceeds token limit
```

**Comment by RussellLuo** [comment]
It might be helpful to change the comment to something like:

```
- model: Name of the model (e.g. `text-davinci-003`)
    This is used to determine the context length, whether it's a chat model, and whether it supports function calling.
```

Additional, I think it's challenging for users to correctly get the full model name as [Azure splits it into two fields `model name` and `model version`](https://learn.microsoft.com/en-us/azure/ai-services/openai/how-to/working-with-models?tabs=powershell#model-updates). All of the models listed in [Azure docs](https://learn.microsoft.com/en-us/azure/ai-services/openai/concepts/models) lack standardized and unified names, which might make users confused.

For example, when viewing the table as below (see [GPT-3.5 models](https://learn.microsoft.com/en-us/azure/ai-services/openai/concepts/models#gpt-35-models) for the full table):

Model ID | Max Request (tokens) | Training Data (up to)
-- | -- | --
gpt-35-turbo (0301) | 4,096 | Sep 2021
gpt-35-turbo (0613) | 4,096 | Sep 2021
... | ... | ...
gpt-35-turbo (1106) | Input: 16,385<br>Output: 4,096 | Sep 2021

I'm not sure which model the default name `"gpt-35-turbo"` refers to. Likewise, I am uncertain which name, `"gpt-35-turbor"` or `"gpt-35-turbo-1106"`, should be used if I want to use model `gpt-35-turbo (1106)`.

I think it might be easier and more intuitive if we can just configure the settings such as `context_window`, `is_chat_model` and `is_function_calling_model` directly.

**Comment by mve** [comment]
Yes the naming of models in Azure can be a bit tricky. But at the same time, wouldn't it be even more tedious for people to find the correct context windows for different models? Also finding the correct context window would still involve users figuring out what model they are using. Also not sure if changing the approach for AzureOpenAI vs OpenAI is a good idea, would make those 2 implementations work quite different right?

For those reasons I personally think it would be better to stick with using the model name. Just needs to be documented better and i.m.o. the default model value needs to be removed to prevent accidentally selecting the wrong model.

**Comment by logan-markewich** [comment]
I'm like 99% sure this is no longer an issue, as I haven't seen anyone note this in quite some time. Please feel free to open a new issue if otherwise
